crates/sui-types/src/execution_mode.rs

```diff
@@ -199,27 +199,27 @@ impl ExecutionMode for System {
 /// WARNING! Using this mode will bypass all normal checks around Move entry functions! This
 /// includes the various rules for function arguments, meaning any object can be created just from
 /// BCS bytes!
-pub struct DevInspect;
+pub struct DevInspect<const SKIP_ALL_CHECKS: bool>;
```

Comment:
I think your life will be easier if this is not a compile-time constant, but a field.

```suggestion
pub struct DevInspect { skipChecks: bool };
```

Reply:
Ah, but I see `ExecutionMode` functions don't accept `self` ...that is annoying. Ok, ignore me!